docs/scos/dev/front-end-development/202307.0/oryx/building-applications/styling/oryx-responsive-design.md

```diff
@@ -0,0 +1,215 @@
+
+In the next sections, we will explore how to leverage Oryx's Responsive Design system to create responsive layouts, adapt individual components, and utilize media queries effectively.
+
+## Build Responsive Designs
+
+There are various techniques involved to apply responsive designs in your pages and components:
+
+- Components can implement different styles per screen size
+- Layouts can implement different styles per screen size
+- Images can have different sizes for optimized experiences per screen size
+- Experience structures can change per screen size
+
+### Responsive Component Styles
+
+CSS allows to differentiate style rules per screen size, using media queries. Media queries can handle various rules to associate css rules to a screen. In the context of responsive design, we're solely concerned with the width of the viewport. The media query rules are unfortunately not configurable, for example by CSS variables. This is why Oryx provides a mechanism to configure styles per screen size, rather than provide hardcoded media query rules inside the styles.
+
+In order to associate styles to a specific screen size, you can link the styles in the component definition. The styles can be lazily loaded, so that irrelevant styles are not loaded.
+
+<!-- link "component definition" to the component docs once published -->
+
+```ts
+export const selectComponent = componentDef({
+  name: "oryx-select",
+  // ...
+  stylesheets: [
+    {
+      rules: () => import("./styles").then((m) => m.screenStyles),
+    },
+  ],
+});
+```
+
+The example above, loads a file (`styles.ts`), that exports `const screenStyles`. The rules are expected to be an array of `CssStylesWithMedia` type, which is exported from the [utilities package](https://www.npmjs.com/package/@spryker-oryx/utilities). The lazy loaded styles can contain either the styles for a single screen size, or various screen sizes.
+
+```ts
+const screenStyles: CssStylesWithMedia[] = [
+  {
+    media: {
+      [DefaultMedia.Screen]: Breakpoint.Sm,
+    },
+    css: css`
+      :host {
+        // put styles for small screens here
+      }
+    `,
+  },
+];
+```
+
+{% info_block infoBox "" %}
+
+When you do not need to align your custom styles with configurable breakpoints in your custom code, as the breakpoints are no likely to change, you can avoid the configurable styles in the component definition and use standard media queries in your stylesheets.
+
+{% endinfo_block %}
+
+## Responsive Layouts
+
+The layout system has build in support for responsive layouts, which are driven by [design tokens](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-applications/styling/oryx-design-tokens.html). Layouts, such as a grid or carousel are designed to render different number of items per row or column, depending on the screen size. The number of items is configurable as a design token.
```

Comment:
Do we need to provide our default grid columns configuration per screen size, ie 12 for `lg`, 4 for `sm`?

Reply:
no, that's part of the layout documentation. It is not yet published, but will be before this one. If you're interested, you can have a read (#2211)